qp_qiime2/qp_qiime2.py

```diff
@@ -268,9 +273,14 @@ def call_qiime2(qclient, job_id, parameters, out_dir):
             key = parameters.pop(k)
             q2inputs[key] = ('', '')
 
-    # if the command_has_where_statement and is not in the q2inputs we need to
-    # remove the metadata file
-    if command_has_where_statement and 'metadata' in q2inputs:
+    # if 'metadata' is in q2inputs but 'where' exist and is empty in q2params,
+    # remove the parameter metadata
+    # NOTE: AFAIK there is no way to differentiate between sample and prep
```

Comment:
Wondering if this means we should be propagating that metadata to this part of the code.

Reply:
Good question, there is no way to do that as there is no feature metadata - at least currently; thus simply ignoring/removing